### ISSUE TITLE
Clarification gets stuck in BEING ANSWERED state 

### ISSUE THREAD

**Issue by johnbrvc** [pc2ccs/pc2v9#493]
Thank you for working to help us improve PC^2!  
Please fill in the following fields in order to help us understand what issue you are reporting.

**Describe the issue**: 
It has been observed on the KSA (ACPC) contest that a clarification was in BEING ANSWERED state when no judge was
answering it.  This led to the clarification being stuck.  Attempts to edit the clarification from Administrators failed since
the Edit Clarification function is not implemented. 

**To Reproduce**: 
Can not reproduce, but the problem was observed.  It may have to do with the fact that it was observed on a multi-site
contest and some of the sites were trying to send to other sites that were not there.  (The sites were removed prior to contest
start, but the server was not restarted).  The logs show the servers can't be reached to send messages related to the
Clarification.  This could be a contributing factor.

**Expected behavior**: 
Clarifications should not be marked as BEING ANSWERED if no on is currently answering it.

**Actual behavior**: 
A clarification was marked as BEING ANSWERED yet no one was answer it.

**Environment**: 
Version 9.8build 20220713 (Wednesday, July 13th 2022 15:07 UTC) build 6258~develop
Java version 11.0.15
OS: Linux 5.13.0-1031-aws (amd64)
Ubuntu 20.04

**Log Info**: 
```
220715 084938.153|INFO|Thread-73648|log|sendToRemoteServer SERVER0 @ site 13 at 13 Packet[CLARIFICATION_SUBMISSION_CONFIRM] #111492 SERVER0 @ site 10 -> SERVER0 @ site 13 null
220715 084938.153|SEVERE|Thread-73648|log|Unable to send packet to site 13 (SERVER0 @ site 13)Packet[CLARIFICATION_SUBMISSION_CONFIRM] #111492 SERVER0 @ site 10 -> SERVER0 @ site 13
220715 084938.153|INFO|Thread-73648|log|sendToRemoteServer SERVER0 @ site 11 at 11 Packet[CLARIFICATION_SUBMISSION_CONFIRM] #111493 SERVER0 @ site 10 -> SERVER0 @ site 11 null
220715 084938.153|SEVERE|Thread-73648|log|Unable to send packet to site 11 (SERVER0 @ site 11)Packet[CLARIFICATION_SUBMISSION_CONFIRM] #111493 SERVER0 @ site 10 -> SERVER0 @ site 11
220715 084938.153|INFO|Thread-73648|log|sendToRemoteServer SERVER0 @ site 12 at 12 Packet[CLARIFICATION_SUBMISSION_CONFIRM] #111494 SERVER0 @ site 10 -> SERVER0 @ site 12 null
220715 084938.154|SEVERE|Thread-73648|log|Unable to send packet to site 12 (SERVER0 @ site 12)Packet[CLARIFICATION_SUBMISSION_CONFIRM] #111494 SERVER0 @ site 10 -> SERVER0 @ site 12
220715 084938.154|INFO|Thread-73648|log|sendToRemoteServer SERVER0 @ site 14 at 14 Packet[CLARIFICATION_SUBMISSION_CONFIRM] #111495 SERVER0 @ site 10 -> SERVER0 @ site 14 null
220715 084938.154|SEVERE|Thread-73648|log|Unable to send packet to site 14 (SERVER0 @ site 14)Packet[CLARIFICATION_SUBMISSION_CONFIRM] #111495 SERVER0 @ site 10 -> SERVER0 @ site 14
220715 084938.154|INFO|Thread-73648|log|sendToRemoteServer SERVER0 @ site 13 at 13 Packet[CLARIFICATION_SUBMISSION_CONFIRM] #111496 SERVER0 @ site 10 -> SERVER0 @ site 13 null
220715 084938.154|SEVERE|Thread-73648|log|Unable to send packet to site 13 (SERVER0 @ site 13)Packet[CLARIFICATION_SUBMISSION_CONFIRM] #111496 SERVER0 @ site 10 -> SERVER0 @ site 13
220715 084938.154|INFO|Thread-73648|log|sendToRemoteServer SERVER0 @ site 12 at 12 Packet[CLARIFICATION_SUBMISSION_CONFIRM] #111497 SERVER0 @ site 10 -> SERVER0 @ site 12 null
220715 084938.154|SEVERE|Thread-73648|log|Unable to send packet to site 12 (SERVER0 @ site 12)Packet[CLARIFICATION_SUBMISSION_CONFIRM] #111497 SERVER0 @ site 10 -> SERVER0 @ site 12
220715 084938.154|INFO|Thread-73648|log|sendToRemoteServer SERVER0 @ site 13 at 13 Packet[CLARIFICATION_SUBMISSION_CONFIRM] #111498 SERVER0 @ site 10 -> SERVER0 @ site 13 null
220715 084938.154|SEVERE|Thread-73648|log|Unable to send packet to site 13 (SERVER0 @ site 13)Packet[CLARIFICATION_SUBMISSION_CONFIRM] #111498 SERVER0 @ site 10 -> SERVER0 @ site 13
```

**Screenshots**:  if applicable, please add attachments showing screenshots to help explain the issue.

**Additional context**: please add any other context about the issue here.

**Comment by clevengr** [comment]
John Buck submitted the following update on this issue, based on the GNY 2023 Regional:
```
Clarifications:  “Being Answered” “hung” status.  We figured out the recipe that makes this happen (at which step can YOU figure out what the issue is?):
2 pending NEW clarifications: 10 and 11
Judge1 selects clar 10, and presses “answer”
Status of clar 10 (on admin view) correctly shows “Being answered”
Judge1 now selects clar 11 (while still answer clar 10).
Got it yet?
Modeless “answer clarification” dialog has its contents replaced with the info for clar 11 (correctly).
Both clarifications 10 and 11 show “being answered” and clar 10 is forever left in that state until judge1 logs out, or the contest is restarted.  This happened 3 times during GNY regional
```

**Comment by clevengr** [comment]
Discussion at a meeting on 11/1/2023 resulted in agreement that a fix for V9.9 should just involve popping up a dialog when a judge attempts to check out a second clar, explaining that it's only allowed to check out one clar at a time, and giving them the option of (automatically) returning their first clar and checking out the second one.
